### ISSUE TITLE
Behavior difference: False positive on plain-js class attribute spread assignment

### ISSUE THREAD

**Issue by hkleungai** [microsoft/typescript-go#3642]
## Steps to reproduce

1. Start a plain new project
    - "@typescript/native-preview": "7.0.0-dev.20260428.1",
    - "typescript": "6.0.3"
2. Run `npx tsgo --init` to get the default tsconfig. Turn on these flags as well.
    - "allowJs": true,
    - "checkJs": true,
    - "stableTypeOrdering": true,
3. Add `src/main.js`. Run `npx tsc --noEmit` & `npx tsgo --noEmit` to see the difference.

```js
// src/main.js
export class C {
    constructor() {
        this.stuffs = {}
    }

    /**
     * @param {Object} stuffs
     */
    addStuffs(stuffs) {
        this.stuffs = { ...stuffs, ...this.stuffs }
    }
}
```

## Behavior with `typescript@6.0`

No error at all.

## Behavior with `tsgo`

```
src/main.js:11:9 - error TS2322: Type '{ constructor: Function; toString(): string; toLocaleString(): string; valueOf(): Object; hasOwnProperty(v: PropertyKey): boolean; isPrototypeOf(v: Object): boolean; propertyIsEnumerable(v: PropertyKey): boolean; }' is not assignable to type '{}'.
  Property 'constructor' does not exist on type '{}'.

11         this.stuffs = { ...stuffs, ...this.stuffs }
           ~~~~~~~~~~~


Found 1 error in src/main.js:11
```

**Comment by ahejlsberg** [comment]
This is caused by the same issue as #3639.
